Mathlib/RingTheory/Coalgebra/Convolution.lean

```diff
@@ -0,0 +1,136 @@
+/-
+Copyright (c) 2025 Yaël Dillies, Michał Mrugała, Yunzhou Xie. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Michał Mrugała, Yunzhou Xie
+-/
+import Mathlib.LinearAlgebra.TensorProduct.Tower
+import Mathlib.RingTheory.Coalgebra.Basic
+
+/-!
+# Convolution product on linear maps from a coalgebra to an algebra
+
+This file constructs the ring structure on linear maps `C → A` where `C` is a coalgebra and `A` an
+algebra, where multiplication is given by `(f * g)(x) = ∑ f x₍₁₎ * g x₍₂₎` in Sweedler notation or
+```
+         |
+         μ
+|   |   / \
+f * g = f g
+|   |   \ /
+         δ
+         |
+```
+diagrammatically, where `μ` stands for multiplication and `δ` for comultiplication.
+-/
+
+suppress_compilation
+
+open Coalgebra TensorProduct
+
+variable {R A B C : Type*} [CommSemiring R]
+
+namespace LinearMap
+
+local notation "η" => Algebra.linearMap R A
+local notation "ε" => counit (R := R) (A := C)
+local notation "μ" => mul' R A
+local notation "δ" => comul
```

Comment:
Can you make a PR that scopes these to `BiAlgebra` or `HopfAlgebra`? I think there's a risk that we invent this over and over again across multiple files.